TrackingTools/MeasurementDet/src/LayerMeasurements.cc

```diff
@@ -30,6 +31,26 @@ namespace {
     result.emplace_back(ts, std::make_shared<InvalidTrackingRecHit>(det, TrackingRecHit::missing), 0.F, &layer);
   }
 
+  // layerIsCompatibleWithTSOS:
+  //  - enforce minimum requirements of compatibility between DetLayer and TrajectoryStateOnSurface
+  //    (e.g. global position of TSOS within detector volume)
+  //  - current implementation does not make use of layer argument,
+  //    see below for an example of how to use it
+  //      #include "Geometry/CommonDetUnit/interface/GeomDetEnumerators.h"
+  //       [..]
+  //      using namespace GeomDetEnumerators;
+  //      auto const& det = layer.subDetector();
+  //      if (det == PixelBarrel or det == PixelEndcap) {
+  //        return xa < 2e1 and ya < 2e1 and za < 6e1;
+  //      }
+  bool layerIsCompatibleWithTSOS(DetLayer const& /* layer */, TrajectoryStateOnSurface const& tsos) {
+    auto const& gp = tsos.globalPosition();
+    auto const xa = std::abs(gp.x());
+    auto const ya = std::abs(gp.y());
+    auto const za = std::abs(gp.z());
+    return xa < 1e3 and ya < 1e3 and za < 1e3;
```

Comment:
These are arguably magic numbers. Suggestions on how to improve this check are welcome.

Reply:
1e3 cm = 10m ? Should we use instead the inner tracking volume? Or is it used also for muons?

Reply:
Indeed, I removed the detector-type check too quickly. I changed it to 12m in z, which should include the muon detectors.